applications/FluidDynamicsApplication/python_scripts/fluid_dynamics_analysis.py

```diff
@@ -0,0 +1,232 @@
+        for i in range(self.project_parameters["initial_conditions_process_list"].size()):
+            initial_cond_part_name = self.project_parameters["initial_conditions_process_list"][i]["Parameters"]["model_part_name"].GetString()
+            self.model.AddModelPart(self.main_model_part.GetSubModelPart(initial_cond_part_name))
+
+        # Add the gravity SubModelParts to the model
+        for i in range(self.project_parameters["gravity"].size()):
+            gravity_part_name = self.project_parameters["gravity"][i]["Parameters"]["model_part_name"].GetString()
+            self.model.AddModelPart(self.main_model_part.GetSubModelPart(gravity_part_name))
+
+    def SetUpConditions(self):
+        '''Read the boundary and initial conditions for the problem and initialize the processes that will manage them.'''
+
+        ## Processes construction
+        from process_factory import KratosProcessFactory
+        factory = KratosProcessFactory(self.model)
+        # The list of processes will contain a list with each individual process already constructed (boundary conditions, initial conditions and gravity)
+        # Note 1: gravity is constructed first. Outlet process might need its information.
+        # Note 2: initial conditions are constructed before BCs. Otherwise, they may overwrite the BCs information.
+        self.simulation_processes =  factory.ConstructListOfProcesses( self.project_parameters["gravity"] )
+        self.simulation_processes += factory.ConstructListOfProcesses( self.project_parameters["initial_conditions_process_list"] )
+        self.simulation_processes += factory.ConstructListOfProcesses( self.project_parameters["boundary_conditions_process_list"] )
+        self.simulation_processes += factory.ConstructListOfProcesses( self.project_parameters["auxiliar_process_list"] )
+
+        for process in self.simulation_processes:
+            process.ExecuteInitialize()
+
+    def SetUpSolution(self):
+        '''Initialize the Python solver and its auxiliary tools and processes.'''
+
+        self.solver.Initialize()
+
+        #TODO this should be generic
+        # initialize GiD  I/O
+        self._SetUpGiDOutput()
+
+        for process in self.simulation_processes:
+            process.ExecuteBeforeSolutionLoop()
+
+        ## Stepping and time settings
+        self.end_time = self.project_parameters["problem_data"]["end_time"].GetDouble()
+
+        ## Writing the full ProjectParameters file before solving
+        if self.is_printing_rank and self.echo_level > 1:
+            with open("ProjectParametersOutput.json", 'w') as parameter_output_file:
+                parameter_output_file.write(self.project_parameters.PrettyPrintJsonString())
+
+
+    def _SetUpGiDOutput(self):
+        '''Initialize self.output as a GiD output instance.'''
+        self.have_output = self.project_parameters.Has("output_configuration")
+        if self.have_output:
+            if self.parallel_type == "OpenMP":
```

Comment:
As a suggestion, the following is slightly shorter and you avoid duplicated code:
~~~py
if (self.parallel_type == "OpenMP"):
    from gid_output_process import GiDOutputProcess as output_process
elif (self.parallel_type == "MPI"):
    from gid_output_process_mpi import GiDOutputProcessMPI as output_process

self.gid_output = output_process(self.solver.GetComputingModelPart(),
                                    self.ProjectParameters["problem_data"]["problem_name"].GetString(),
                                    self.ProjectParameters["output_configuration"])
~~~

Reply:
Nice trick, will do.